quickstart-28: richfaces-validation

lfryc opened the pull request:
based upon quickstart-27: bean-validation

Commit:
initial commit of quickstart-28: richfaces-validation (based upon quickstart-27)

pmuir commented on richfaces-validation/README.md:
```diff
@@ -0,0 +1,193 @@
+get your foot in the door developing with Java EE 6 on JBoss AS 7 or JBoss Enterprise Application Platform 6. 
+This project is setup to allow you to use JSF 2.0, RichFaces 4.2, CDI 1.0, JPA 2.0 and Bean Validation 1.0. 
+
+It consists from one entity - Member - which is annotated with JSR-303 (Bean Validation)
```
"consists of" not "consists from"

pmuir commented on richfaces-validation/README.md:
```diff
@@ -0,0 +1,193 @@
+This project is setup to allow you to use JSF 2.0, RichFaces 4.2, CDI 1.0, JPA 2.0 and Bean Validation 1.0. 
+
+It consists from one entity - Member - which is annotated with JSR-303 (Bean Validation)
+constraints. These constraints are in typical applications checked on several places:
```
Missing commas "These constraints are, in typical applications, checked on several places:" and also s/on/in/

pmuir commented on richfaces-validation/README.md:
```diff
@@ -0,0 +1,193 @@
+
+* as database constraints
+* on the persistence layer
+* once entity enters view layer (using JSF / Bean Validation integration)
```
probably clearer just to say "the view layer"

pmuir commented on richfaces-validation/README.md:
```diff
@@ -0,0 +1,193 @@
+* once entity enters view layer (using JSF / Bean Validation integration)
+* on the client side (using RichFaces 4.2 - Client Side Validation)
+
+That all in the "define once - check in each layer" manner.
```
I think you mean to discuss the advantage that this offers - DRY. Perhaps rephrase in these terms?

"This allows you to not repeat the definitions of the constraints, but still have them applied at all layers, and thus report good error messages to users"

pmuir commented on richfaces-validation/README.md:
```diff
@@ -0,0 +1,193 @@
+That all in the "define once - check in each layer" manner.
+
+
+However this sample does not contain any persistence layer, since it shows mainly integration
```
s/sample/quickstart/

pmuir replied:
s/mainly//

pmuir commented on richfaces-validation/README.md:
```diff
@@ -0,0 +1,193 @@
+However this sample does not contain any persistence layer, since it shows mainly integration
+of RichFaces, JSF and Bean Validation.
+
+There are tests for Bean Validation constraints for Member entity which allows you to check
```
s/for/for all/

pmuir commented on richfaces-validation/README.md:
```diff
@@ -0,0 +1,193 @@
+of RichFaces, JSF and Bean Validation.
+
+There are tests for Bean Validation constraints for Member entity which allows you to check
+implementation without necessity to check the application. These tests are written using Arquillian.
```
I'm not sure what you mean to say here.

pmuir commented on richfaces-validation/README.md:
```diff
@@ -0,0 +1,193 @@
+
+* e-mail
+* name and phone
+* password and its confirmation
```
Can you remove password, as we don't want to recommend this approach of storing user passwords to our users.

pmuir commented on richfaces-validation/README.md:
```diff
@@ -0,0 +1,193 @@
+* password and its confirmation
+* confirmation of all inserted data
+
+Once user is successfully registered, he is redirected to initial page with message
```
s/he is/they are/

pmuir replied:
s/with message/with the message/

pmuir commented on richfaces-validation/README.md:
```diff
@@ -0,0 +1,193 @@
+* confirmation of all inserted data
+
+Once user is successfully registered, he is redirected to initial page with message
+that he has been successfully registered.
```
s/that he has been successfully registered/that they have successfully registered/

pmuir commented on richfaces-validation/README.md:
```diff
@@ -0,0 +1,193 @@
+Building WAR deployment with Maven
+===============================
+
+To build the application, the only you need is trigger Maven build from
```
This doesn't really read well in English.

pmuir commented on richfaces-validation/README.md:
```diff
@@ -0,0 +1,193 @@
+Deploying the WAR to the JBoss AS
+=================================
+
+To deploy the application to JBoss AS 7, you needs just copy the built WAR
```
We don't use this approach in the quickstarts, so can you remove it?

pmuir commented on richfaces-validation/README.md:
```diff
@@ -0,0 +1,193 @@
+    
+In this case, you should see following output in server console:
+
+    JBAS018558: Undeployed "richfaces-validation.war"
```
I would suggest replacing this with the standard text.

pmuir commented on richfaces-validation/README.md:
```diff
@@ -0,0 +1,193 @@
+managed or remote container, the difference is that the remote one requires a running JBoss AS 7 / 
+JBoss Enterprise Application Platform 6 instance prior executing tests.
+
+**Testing on Remote Server**
```
Can you use markdown syntax here?

pmuir commented on richfaces-validation/pom.xml:
```diff
@@ -0,0 +1,258 @@
+            <scope>provided</scope>
+        </dependency>
+        
+        <!-- Import RichFaces runtime dependencies -->
```
Might be worth mentioning that you do include this as a library in the war.

pmuir commented on richfaces-validation/pom.xml:
```diff
@@ -0,0 +1,258 @@
+                        mvn package jboss-as:deploy -->
+                    <plugin>
+                        <groupId>org.jboss.as.plugins</groupId>
+                        <artifactId>jboss-as-maven-plugin</artifactId>
```
Shouldn't this be in <plugin> in the main build?

pmuir commented on richfaces-validation/src/main/java/org/jboss/as/quickstarts/kitchensink/web/RegistrationBean.java:
```diff
@@ -0,0 +1,88 @@
+    transient Flash flash;
+
+    @Inject
+    transient FacesContext facesContext;
```
This won't work, as CDI won't reinject on reserialization. Need to find a better way to do this (e.g. inject Instance)

pmuir commented:
Lukas did you manage to address my comments?

lfryc commented:
Pete, I have run into issues having unexpected conference engagement.
What is next planned released I shoul target?

Commit:
incorporated comments/fixes from https://github.com/jbossas/quickstart/pull/122

Commit:
removed password from registration wizard + polished

lfryc commented:
Hi Pete, I have been able to incorporate all comments in README and removed password from registration wizard.

sgilda commented:
Hi Lukas,
I'm testing your quickstart for usability. It builds and deploys great, however, I ran into a couple of problems and have a few comments.

-- Could you add the command line to deploy to the quickstart README? I'm usually in a hurry and don't use an IDE and I'm guessing there are others out there like me. ;-)
           mvn jboss-as:deploy
-- Could you change the web context from richfaces-validation to jboss-as-richfaces-validation to be consistent with the other quickstarts? It also assures this one will be unique.
-- I was playing around with validation. 
   ---- When I entered 'sgilda' and clicked on "Continue", it gave an error "not an well-formed e-mail". So far, so good.
   ---- Then I entered 'sgilda@red' and clicked on "Continue". It cleared up the errors, but stayed on the same screen. Shouldn't it go to the next screen?
   ---- I clicked continue once more and it this time it took me to the next screen.
   ---- On the Registration form page, I decided to click on "Back" to go back and append the ".com" to my email address. It gave me validation errors for both fields. It shows Name: "size must be between 1 and 25" and "Phone number must be 10-12 digits". Should it be performing validation when I click the "Back" button?

Thanks, Sande

Commit:
richfaces-validation: added intstructions to readme and changed context root

Commit:
moved rich:messages behind the form

lfryc commented:
Hello Sande, I have addressed all the issues you mentioned, except the last one:

"---- On the Registration form page, I decided to click on "Back" to go back and append the ".com" to my email address. It gave me validation errors for both fields. It shows Name: "size must be between 1 and 25" and "Phone number must be 10-12 digits". Should it be performing validation when I click the "Back" button?"

Ideally it should work so, but we have run there into issue with the RichFaces <rich:toggleControl /> component.
It will be addressed in upcoming releases:

https://issues.jboss.org/browse/RF-12068

Since there is no simple workaround,
we have two options: rework it to not use this component
or accept it as it is.

sgilda commented:
Well, it's ugly, but not worth the effort since it's a bug in the underlying component. 

Pete, do you agree? 

I'll check out the changes.

Commit:
polished usage of messages

sgilda commented:
Looks great! I have a few minor nit-picking comments. 
In the README, could you change the following:
Change the title line to read: richfaces-validation: RichFaces and Bean Validation
Change the access URL to a link so when they click on it, it takes them to your running application
    http://localhost:8080/jboss-as-richfaces-validation/

Other than that, this is good to go. Thanks!

Pete, do you need to review anything?

Commit:
fixed title + changed the application context URLs to links

pmuir commented:
Made updates and merged.

lfryc commented:
Great, thanks Pete, Sande.
